pkg/kubelet/nodestatus/setters.go

```diff
@@ -197,36 +197,40 @@ func NodeAddress(nodeIP net.IP, // typically Kubelet.nodeIP
 			// For steps 3 and 4, IPv4 addresses are preferred to IPv6 addresses
 			// unless nodeIP is "::", in which case it is reversed.
 			if nodeIPSpecified {
-				ipAddr = nodeIP
+				ipAddr = []net.IP{nodeIP}
 			} else if addr := net.ParseIP(hostname); addr != nil {
-				ipAddr = addr
+				ipAddr = []net.IP{addr}
 			} else {
 				var addrs []net.IP
 				addrs, _ = net.LookupIP(node.Name)
 				for _, addr := range addrs {
 					if err = validateNodeIPFunc(addr); err == nil {
-						if isPreferredIPFamily(addr) {
-							ipAddr = addr
-							break
-						} else if ipAddr == nil {
-							ipAddr = addr
-						}
+						ipAddr = append(ipAddr, addr)
```

Comment:
dual stack is in alpha, I think that this should be used only if the feature gate is enabled
`if utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack)`

Reply:
Good point, thanks.